config/routes.rb

```diff
@@ -22,14 +22,11 @@
   delete 'logout', to: 'sessions#destroy'
 
   # visitable pages
-  resources :users, only: %i[new create show] do
-    resource :profile, except: %i[destroy new] do
-      collection do
-        get :posts
-        get :friends
-        get :media
-      end
-    end
+  resources :users, only: %i[new create show]
+  resource :profile, path: '/profile/:unique_name', except: %i[destroy new] do
+    get :posts, on: :member
+    get :friends, on: :member
+    get :media, on: :member
```

Comment:
`localhost:5000/profile/emmahyde` works now!

Reply:
Nice!!

Yea, I didn't enjoy the nested resource paths.